flink-runtime/src/main/java/org/apache/flink/runtime/state/IncrementalRemoteKeyedStateHandle.java

```diff
@@ -361,9 +362,10 @@ IncrementalRemoteKeyedStateHandle copy() {
 
     /** Create a unique key to register one of our shared state handles. */
     @VisibleForTesting
-    public SharedStateRegistryKey createSharedStateRegistryKeyFromFileName(StateHandleID shId) {
+    public SharedStateRegistryKey createSharedStateRegistryKey(StreamStateHandle handle) {
+        String keyString = handle.getStreamStateHandleID().getKeyString();
         return new SharedStateRegistryKey(
-                String.valueOf(backendIdentifier) + '-' + keyGroupRange, shId);
+                UUID.nameUUIDFromBytes(keyString.getBytes(StandardCharsets.UTF_8)).toString());
```

Comment:
This won't work when multiple handles point to the same file (but use different offsets), right?
Should we switch to using completely random IDs (not related to remote nor local file paths)?

Reply:
Do you mean like in ChangelogStateHandleStreamImpl ? 
Since the basic unit of deletion is the StreamStateHandle, I instead think it's correct to use a key based on StreamStateHandle#getStreamStateHandleID() . If SharedStateRegisterKey is based on <StreamStateHandle,offset> pairs, files may be deleted while they are still in use.
And the changes here do not affect ChangelogStateHandleStreamImpl, whose registration logic is independent.

Reply:
Yes, I think you're right.